### PR TITLE
Add StringBuilderExtensions, consolidate StringBuilder helpers

### PR DESCRIPTION
**Pull request by ivandrofly**
## Summary
- New `Nikse.SubtitleEdit.Core.Common.StringBuilderExtensions` class consolidating generic StringBuilder helpers that were scattered as private/protected statics
- `Trim()` — moved from `SubtitleFormat.TrimBuilder` (in-place trim, avoids the `sb.ToString().Trim()` extra allocation); call sites in `SubRip` and `AdvancedSubStationAlpha` updated
- `StartsWith(char)` / `EndsWith(char)` — moved (not copied, to avoid ambiguous-call errors) from `StringExtensions`; existing call sites unaffected since the namespace is unchanged
- `CountChar(char)` — moved from `MergeAndSplitHelper`; `StringBuilder.GetChunks()` turned out to be missing from the netstandard2.1 reference assemblies, so that target falls back to an indexer loop behind the existing `#if NET8_0_OR_GREATER` guard pattern
- `AppendNumber(int, int)` — moved from `AdvancedSubStationAlpha` (zero-padded append matching `"{0:00}"`)
- Adds `StringBuilderExtensionsTest` (21 tests), including a cross-chunk `CountChar` case
- Behavior is unchanged; bodies were moved verbatim. Format-specific helpers (`AppendTimeCode`, EBU/MP4 tag helpers) stay where they are

## Test plan
- [ ] `dotnet build src/libse/LibSE.csproj` — both netstandard2.1 and net10.0 targets compile
- [ ] `dotnet test tests/libse/LibSETests.csproj` — full suite passes (956/956 locally)
- [ ] Save a subtitle as SubRip and as ASSA and confirm output is unchanged (leading/trailing whitespace trimmed, ASSA time codes still `h:mm:ss.cc`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/libse/Common/StringBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                value = -value;
+            }
+
+            if (minDigits >= 2 && value < 10)
+            {
+                sb.Append('0');
+            }
+
+            sb.Append(value);
+        }
```
[the remark this answers is not in the document]
Comment (ivandrofly) [reply src/libse/Common/StringBuilderExtensions.cs:93]: Fixed in cfe40e4 — AppendNumber now pads correctly to minDigits and negates via long so int.MinValue does not overflow.

File: tests/libse/Core/StringBuilderExtensionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    [Fact]
+    public void AppendNumberNegative()
+    {
+        var sb = new StringBuilder();
+        sb.AppendNumber(-5, 2);
+        Assert.Equal("-05", sb.ToString());
+    }
+}
```
[the remark this answers is not in the document]
Comment (ivandrofly) [reply tests/libse/Core/StringBuilderExtensionsTest.cs:186]: Edge-case tests added in cfe40e4 (int.MinValue and minDigits=3 padding).
